### PR TITLE
[ACS-3548] Accessibility issues

### PR DESCRIPTION
**Pull request by jatin2008**
**Please check if the PR fulfills these requirements**

> - [ x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
Accessibility issues present

**What is the new behaviour?**
Accessibility issues fixed

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

File: lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html [hunk 1]
```diff
@@ -5,7 +5,7 @@
                              [expanded]="canExpandProperties()"
                              [attr.data-automation-id]="'adf-metadata-group-properties'">
             <mat-expansion-panel-header>
-                <mat-panel-title role="heading">
+                <mat-panel-title role="heading" aria-level="3">
```
Comment (AleksanderSklorz) [inline lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html:8]: I see here still some places which are using aria-level, can you use h1, h2 etc. everywhere instead of aria-level in that PR? I think it should be possible if you do it like: 
<mat-panel-title>
  <h3>{{ 'CORE.METADATA.BASIC.HEADER' | translate }}</h3>
</mat-panel-title>

**Comment by MichalKinas** [comment]
What about 879018?

File: lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html (hunk 1 above)
Comment (jatin2008) [reply lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html]: I resolved with heading tags instead of div

Comment (AleksanderSklorz) [reply lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html]: Thank you. @jatin2008 I think you can also remove now these role"heading" right? No need to say that it is heading if you use h1/h2 etc. which already mean "heading"

File: lib/core/src/lib/datatable/components/datatable/datatable.component.html [hunk 2]
```diff
@@ -1,4 +1,4 @@
-<div
+k<div
```
Comment (mauriziovitale) [inline lib/core/src/lib/datatable/components/datatable/datatable.component.html:1]: there is a k typo

**Review by mauriziovitale — DISMISSED** [review]
check the comment

File: lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html (hunk 1 above)
Comment (jatin2008) [reply lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html]: noted and resolved

File: lib/core/src/lib/datatable/components/datatable/datatable.component.html (hunk 2 above)
Comment (jatin2008) [reply lib/core/src/lib/datatable/components/datatable/datatable.component.html:1]: noted and resolved

**Comment by jatin2008** [comment]
> What about 879018?

when I select a particular item in the list it reads out the name of the item, along with it reads selected and also 1 of 21, so it is working as expected.

**Comment by jatin2008** [comment]
> check the comment

noted and resolved

**Comment by jatin2008** [comment]
> check the comment

noted and resolved

**Comment by jatin2008** [comment]
> check the comment

noted and resolved

File: lib/content-services/src/lib/breadcrumb/breadcrumb.component.html [hunk 3]
```diff
@@ -55,7 +55,7 @@
             {{ item.name | translate }}
         </a>
 
-        <div *ngSwitchDefault class="adf-breadcrumb-item-current" role="heading" aria-level="2"
+        <div *ngSwitchDefault class="adf-breadcrumb-item-current"
```
Comment (MichalKinas) [inline lib/content-services/src/lib/breadcrumb/breadcrumb.component.html:58]: Why did you remove this role?

File: lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html [hunk 4]
```diff
@@ -6,7 +6,7 @@
                              [attr.data-automation-id]="'adf-metadata-group-properties'">
             <mat-expansion-panel-header>
                 <mat-panel-title>
-                    <h1 class="adf-metadata-properties-title">{{ 'CORE.METADATA.BASIC.HEADER' | translate }}</h1>
+                    <h3 class="adf-metadata-properties-title">{{ 'CORE.METADATA.BASIC.HEADER' | translate }}</h3>
```
Comment (MichalKinas) [inline lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html:9]: This will create another accessibility issue, all h tags needs to be in order so if h3 can't be used if h1 and h2 are not present in the DOM tree

File: lib/core/src/lib/comments/comments.component.html [hunk 5]
```diff
@@ -1,7 +1,7 @@
 <div class="adf-comments-container">
-    <h1 id="comment-header" class="adf-comments-header">
+    <h3 id="comment-header" class="adf-comments-header" role="heading">
```
Comment (MichalKinas) [inline lib/core/src/lib/comments/comments.component.html:2]: Same here for h3 tag

**Review by MichalKinas — CHANGES_REQUESTED** [review]

File: lib/content-services/src/lib/breadcrumb/breadcrumb.component.html (hunk 3 above)
Comment (jatin2008) [reply lib/content-services/src/lib/breadcrumb/breadcrumb.component.html]: role and aria-level added again, it was giving some error i think at that time.

File: lib/core/src/lib/comments/comments.component.html (hunk 5 above)
Comment (MichalKinas) [reply lib/core/src/lib/comments/comments.component.html:2]: Btw h tag shouldn't have `heading` role as the tag implies it itself

File: lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html (hunk 4 above)
Comment (jatin2008) [reply lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html]: <h1> tag changed to <h2> as per the discussion as <h1> is already present at the top level

Comment (jatin2008) [reply lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html]: h1 tag changed to h2 as per the discussion as h1 is already present at the top level.

File: lib/core/src/lib/comments/comments.component.html (hunk 5 above)
Comment (jatin2008) [reply lib/core/src/lib/comments/comments.component.html]: h1 tag changed to h2 as per the discussion as h1 is already present at the top level and also heading role removed.

**Review by MichalKinas — APPROVED** [review]

File: lib/content-services/src/lib/permission-manager/components/add-permission/add-permission-panel.component.html [hunk 6]
```diff
@@ -34,7 +35,9 @@
             class="adf-permission-result-list"
             [class.adf-permission-result-list-search]="searchedWord.length === 0">
     <ng-template let-data>
-        <mat-selection-list class="adf-permission-result-list-elements" (keydown.control.a)="onSelectionChange()"
+        <mat-selection-list class="adf-permission-result-list-elements"
+                            title="{{'PERMISSION_MANAGER.ADD-PERMISSION.USER-GROUP-LIST' | translate}}"
+                            (keydown.control.a)="onSelectionChange()"
```
Comment (DenysVuika) [inline lib/content-services/src/lib/permission-manager/components/add-permission/add-permission-panel.component.html:40]: why are you binding Ctrl+a here?

Comment (DenysVuika) [inline lib/content-services/src/lib/permission-manager/components/add-permission/add-permission-panel.component.html:40]: can you see if this one can be removed? binding "ctrl+a" makes no sense here, and it's the only place ever used

File: lib/content-services/src/lib/permission-manager/components/permission-list/permission-list.component.html [hunk 7]
```diff
@@ -73,6 +74,7 @@ <h3>{{'PERMISSION_MANAGER.LABELS.DIRECT-PERMISSIONS' | translate }}</h3>
                 <button
                     mat-button
                     [matTooltip]="'PERMISSION_MANAGER.ACTION.ADD-PERMISSION' | translate"
+                    title="{{'PERMISSION_MANAGER.ACTION.ADD-PERMISSION' | translate}}"
```
Comment (DenysVuika) [inline lib/content-services/src/lib/permission-manager/components/permission-list/permission-list.component.html:77]: this might be conflicting with the matTooltip above... probably remove one of those

File: lib/content-services/src/lib/permission-manager/components/permission-list/permission-list.component.html [hunk 8]
```diff
@@ -81,6 +83,7 @@ <h3>{{'PERMISSION_MANAGER.LABELS.DIRECT-PERMISSIONS' | translate }}</h3>
                 <button
                     mat-button
                     [matTooltip]="'PERMISSION_MANAGER.ACTION.DELETE' | translate"
+                    title="{{'PERMISSION_MANAGER.ACTION.DELETE' | translate}}"
```
Comment (DenysVuika) [inline lib/content-services/src/lib/permission-manager/components/permission-list/permission-list.component.html:86]: conflicts with the matTooltip, one of those should be removed

File: lib/core/src/lib/comments/comments.component.html [hunk 9]
```diff
@@ -1,7 +1,7 @@
 <div class="adf-comments-container">
-    <h1 id="comment-header" class="adf-comments-header">
+    <h2 id="comment-header" class="adf-comments-header">
```
Comment (DenysVuika) [inline lib/core/src/lib/comments/comments.component.html:2]: I'm still not sure what this change is fixing? according to the semantics, h2 should be declared after h1, but h1 is missing here

**Review by DenysVuika — CHANGES_REQUESTED** [review]

File: lib/content-services/src/lib/permission-manager/components/permission-list/permission-list.component.html (hunk 7 above)
Comment (jatin2008) [reply lib/content-services/src/lib/permission-manager/components/permission-list/permission-list.component.html:76]: noted and resolved

File: lib/content-services/src/lib/permission-manager/components/permission-list/permission-list.component.html (hunk 8 above)
Comment (jatin2008) [reply lib/content-services/src/lib/permission-manager/components/permission-list/permission-list.component.html:84]: noted and resolved

**Comment by jatin2008** [comment]
> check the comment

resolved

**Review by nikita-web-ua — APPROVED** [review]

File: lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html [hunk 10]
```diff
@@ -6,7 +6,7 @@
                              [attr.data-automation-id]="'adf-metadata-group-properties'">
             <mat-expansion-panel-header>
                 <mat-panel-title>
-                    <h1 class="adf-metadata-properties-title">{{ 'CORE.METADATA.BASIC.HEADER' | translate }}</h1>
+                    <h2 class="adf-metadata-properties-title">{{ 'CORE.METADATA.BASIC.HEADER' | translate }}</h2>
```
Comment (DenysVuika) [inline lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.html:9]: please use the regular text here and move the class to the parent element, for example:

```
<mat-panel-title class="adf-metadata-properties-title">
{{ 'CORE.METADATA.BASIC.HEADER' | translate }}
</mat-panel-title>
```

File: lib/core/src/lib/comments/comments.component.html (hunk 9 above)
Comment (DenysVuika) [inline lib/core/src/lib/comments/comments.component.html:2]: please use a regular div and add the required styles to the `adf-comments-header` if there's anything needed for the font size or make it bold, that will solve the problems with h1/h2

**Review by DenysVuika — CHANGES_REQUESTED** [review]

**Review by DenysVuika — APPROVED** [review]
